homeassistant/components/homewizard/switch.py

```diff
@@ -84,15 +85,16 @@ async def async_setup_entry(
     """Set up switches."""
     coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
 
-    async_add_entities(
-        HomeWizardSwitchEntity(
-            coordinator=coordinator,
-            description=description,
-            entry=entry,
+    if coordinator.data.device.product_type in SUPPORTS_STATE:
```

Comment:
Since you're checking it so often, would it be an idea to add this as a property on the coordinator ?